akka-actor/src/main/scala/akka/actor/CoordinatedShutdown.scala

```diff
@@ -110,6 +109,9 @@ object CoordinatedShutdown extends ExtensionId[CoordinatedShutdown] with Extensi
     val coord = new CoordinatedShutdown(system, phases)
     initPhaseActorSystemTerminate(system, conf, coord)
     initJvmHook(system, conf, coord)
+    system.registerOnTermination {
+      if (!runningJvmHook) coord.removeJvmShutdownHooks()
+    }
```

Comment:
Is this really the behavior we want? If you do a clean shutdown the actor system will be terminated first and finally the JVM, and also in that case you want your jvm hooks to be run.

Reply:
I'm not quite sure. On one hand that makes sense, on the other hand you do add your shutdown hooks through an extension, which to me implies it is tied to the lifecycle of the extension and the actor system that belongs to. If you want a shutdown hook that has nothing to do with the actor system lifecycle you can just use the JDK/Scala standard lib methods for registering shutdown hooks, no?

Reply:
the original intention with this was to make sure such hooks ran before the artery hooks, otherwise one can use JDK methods.

if we would remove registered hooks on a clean system termination there is no way to have application hooks that guaranteed to be run before artery hooks

Reply:
Hmm, ok, didn't remember that. In that case I guess we should only keep track of, and remove, the `initJvmHook` introduced hook.

Reply:
and perhaps clarify the purpose in the api docs